kbase-extension/static/kbase/js/widgets/appWidgets2/view/newObjectView.js

```diff
@@ -1,57 +1,42 @@
-define([
-    'bluebird',
-    'kb_common/html',
-    '../validation',
-    'common/events',
-    'common/runtime',
-    'common/dom',
-    'bootstrap',
-    'css!font-awesome',
-], (Promise, html, Validation, Events, Runtime, Dom) => {
+define(['bluebird', 'kb_common/html', 'common/dom', 'bootstrap', 'css!font-awesome'], (
+    Promise,
+    html,
+    Dom
+) => {
     'use strict';
 
     const t = html.tag,
         div = t('div'),
         input = t('input');
 
     function factory(config) {
-        let options = {},
+        const options = {},
             spec = config.parameterSpec,
-            parent,
-            container,
             bus = config.bus,
             model = {
-                value: undefined,
-            },
-            dom;
+                value: config.initialValue ? config.initialValue : undefined,
+            };
+        let parent, container, dom;
 
         options.enabled = true;
 
         function setModelValue(value) {
-            return Promise.try(() => {
-                if (model.value !== value) {
-                    model.value = value;
-                    return true;
-                }
-                return false;
-            }).then((changed) => {
-                render();
-            });
+            if (model.value !== value) {
+                model.value = value;
+            }
```

Comment:
may as well just do `model.value = value` and not bother with the equality check